### ISSUE TITLE
Mac release not working on install

### ISSUE THREAD

**Issue by rsterbin** [elwinar/rambler#49]
After a bit of fuss trying to convince Big Sur that it was safe to execute the downloaded file, I managed to get it installed, but it's not working as expected, and the error message leaves something to be desired:

After installing, I tried to get something like a help message:

```
0 [rsterbin@adelle ~]$ rambler
unable to initialize the migration service: directory  unavailable: stat : no such file or directory
```

A few years ago I'd set up a project (with rambler from installed from source), and while I don't remember whether there was a usage/error/help message, I do remember that I need to give it a config file. Maybe that was the problem? Nope:

```
0 [rsterbin@adelle ~/git/stmvirgin/lituricalendar/db/migrations main]$ cat config.json
{
	"directory": ".",
	"driver": "postgres",
	"host": "localhost",
	"port": 5432,
	"user": "rsterbin",
	"password": "xxredactedxx",
	"database": "liturgicalendar",
	"table": "migrations"
}
0 [rsterbin@adelle ~/git/stmvirgin/lituricalendar/db/migrations main]$ ls
20160810131200_schema_services.sql          20160810133220_stmarys_seasons.sql          20160811165200_schema_completed.sql         20170204161816_schema_overrides.sql
20160810131215_stmarys_services.sql         20160810134300_schema_fixed_feasts.sql      20170108223042_schema_floating_feasts.sql   20170204162938_stmarys_overrides.sql
20160810132300_schema_precedence.sql        20160810134357_stmarys_fixed_feasts.sql     20170108223353_stmarys_floating_feasts.sql  20170225143814_schema_calculated.sql
20160810132316_stmarys_precedence.sql       20160810135100_schema_moveable_feasts.sql   20170127204345_schema_federal_holidays.sql  config.json
20160810133200_schema_seasons.sql           20160810135105_stmarys_moveable_feasts.sql  20170127204836_stmarys_federal_holidays.sql
0 [rsterbin@adelle ~/git/stmvirgin/lituricalendar/db/migrations main]$ rambler
unable to initialize the migration service: directory  unavailable: stat : no such file or directory
0 [rsterbin@adelle ~/git/stmvirgin/lituricalendar/db/migrations main]$ rambler apply
unable to initialize the migration service: directory  unavailable: stat : no such file or directory
```

So I threw the release away and installed from source instead:

```
0 [rsterbin@adelle ~]$ go install github.com/elwinar/rambler
go install: version is required when current directory is not in a module
	Try 'go install github.com/elwinar/rambler@latest' to install the latest version
0 [rsterbin@adelle ~]$ go install github.com/elwinar/rambler@latest
```

(skipping install output, it's long and it worked)

Trying again:

```
0 [rsterbin@adelle ~]$ rambler
unable to load configuration from file: open rambler.json: no such file or directory
```

That's an error I can do something about:

```
0 [rsterbin@adelle ~]$ cd ~/git/stmvirgin/lituricalendar/db/migrations
0 [rsterbin@adelle ~/git/stmvirgin/lituricalendar/db/migrations main]$ mv config.json rambler.json
0 [rsterbin@adelle ~/git/stmvirgin/lituricalendar/db/migrations main]$ rambler
NAME:
   rambler - Migrate all the things!

USAGE:
   rambler [global options] command [command options] [arguments...]

AUTHOR:
   Romain Baugue <romain.baugue@elwinar.com>

COMMANDS:
   apply    apply the next migration
   reverse  reverse the last migration
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --configuration value, -c value  path to the configuration file (default: "rambler.json")
   --environment value, -e value    set the working environment (default: "default")
   --debug                          display debug messages
   --help, -h                       show help
```

tl;dr installing the **rambler-darwin-10.6-amd64** release on OSX 11.4 doesn't seem to be working, although installing from source works just fine.

**Comment by elwinar** [comment]
Hi there. Sorry for not answering right away, I was in vacations in a place where the network is unreliable. I got the notification but couldn't load the message completely, not even talking about answering it :D .

The first error message is there because there is no configuration file, so the bootstrap process tries to take values from the environment, and end-up with an empty `directory` flag value, hence the error.

I think `config.json` was the name of the default configuration file in an earlier major version (the project start to be old, we're probably talking about years here, I didn't check in the backlog because it wouldn't bring any useful information), so without using the `configuration` flag it won't be found, and here is the second error.

Last version effectively use `rambler.json` as the default configuration file, and you ended up having the tool start correctly in your last try. I think you're going to find out that the release binary works the same if you provide it the same `rambler.json` configuration file. 

Finally, the real issue here is that the configuration detection is done before displaying and eventual help message, so bootstraping a new project from scratch isn't necessarily straight-forward. I'll try to improve this somehow, because the very first line your tried (naked call without configuration) should answer something helpful instead of a cryptic error message.

**Comment by rsterbin** [comment]
Thanks for getting back to me, and no worries!  When I installed from source, the message was helpful; it was only the release that gave me the cryptic one.

**Comment by elwinar** [comment]
Did you try the release binary with a `rambler.json` configuration file?

**Comment by rsterbin** [comment]
No, I didn't.  I think I deleted the release binary, but I can download again and try it out later today.

**Comment by rsterbin** [comment]
Yep, looks like it works great if it can find a `rambler.json` -- 

```
0 [rsterbin@adelle ~/git/stmvirgin/lituricalendar/db main]$ ~/Downloads/rambler-darwin-10.6-amd64
NAME:
   rambler - Migrate all the things!

USAGE:
   rambler-darwin-10.6-amd64 [global options] command [command options] [arguments...]

VERSION:
   v5.4.0

AUTHOR:
   Romain Baugue <romain.baugue@elwinar.com>

COMMANDS:
   apply    apply the next migration
   reverse  reverse the last migration
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --configuration value, -c value  path to the configuration file (default: "rambler.json")
   --environment value, -e value    set the working environment (default: "default")
   --debug                          display debug messages
   --dry-run                        don't execute any statement, only display them
   --help, -h                       show help
   --version, -v                    print the version
```
